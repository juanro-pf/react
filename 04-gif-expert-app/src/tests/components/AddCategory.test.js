import React from 'react';
const { shallow } = require("enzyme");
const { AddCategory } = require("../../components/AddCategory");
import '@testing-library/jest-dom';

describe('AdCategory', () => {

    const setCategories= jest.fn();
    let wrapper= shallow( <AddCategory setCategories={ setCategories }/>);
    
    beforeEach( () => {
        jest.clearAllMocks;
        wrapper= shallow( <AddCategory setCategories={ setCategories }/>);
    });

    test('Snapshot', () => {
        
        expect(wrapper).toMatchSnapshot();

    });

    test('change text box', () => {
        
        const input= wrapper.find('input');
        const value= 'Hola mundo';
        
        input.simulate('change', { target: {value} });
        
        const inputValue= wrapper.find('p').text().trim();

        expect(inputValue).toBe(value);

    });
    
    test('no postear info onSubmit', () => {
        
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();
        
    });

    test('lamar el setCategories y limpiar la text box', () => {

        const value= 'Hola mundo';
        
        wrapper.find('input').simulate('change', { target: {value} });
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        // expect( setCategories ).toHaveBeenCalled();
        expect( setCategories ).toHaveBeenCalledTimes(1);
        expect( setCategories ).toHaveBeenCalledWith( expect.any(Function) ); // Haberlo llamado con una funci√≥n como parametro
        
        const afterValue= wrapper.find('input').prop('value');

        expect(afterValue).toBe('');
        
    });
    
    
});