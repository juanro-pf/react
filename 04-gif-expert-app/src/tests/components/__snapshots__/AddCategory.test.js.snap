// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdCategory Snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Nuevo Anime"
    type="text"
    value=""
  />
</form>
`;
